model-client/src/commonMain/kotlin/org/modelix/model/client2/ModelClientV2.kt

```diff
@@ -165,6 +165,18 @@ class ModelClientV2(
         return receivedVersion
     }
 
+    override suspend fun pullHash(branch: BranchReference): String {
+        val response = httpClient.get {
+            url {
+                takeFrom(baseUrl)
+                appendPathSegmentsEncodingSlash("repositories", branch.repositoryId.id, "branches", branch.branchName, "hash")
```

Comment:
This repeats throughout the different methods implemented in this class (apart from the last argument). Refactoring the path creation for a repo and branch into a separate method would be good.